### ISSUE TITLE
False positive of `return in a defer function has no effect`

### ISSUE THREAD

**Issue by rpstw** [mgechev/revive#1029]
**Describe the bug**
Unexpected `return in a defer function has no effect`

**To Reproduce**
1. use the 1.3.9 version
2. use this go code
```go
package main

func verify(a any) func() {
	return func() {
		fn := a.(func() error)
		if err := fn(); err != nil {
			panic(err)
		}
	}

}

func main() {
	defer verify(func() error {
		return nil
	})()
}
```

3. use this config
As of my understanding, it disable the `call-chain` defer check but enable the `return` defer check, which is our case.
```toml
[rule.defer]
  arguments=[["return"]]
```

**Expected behavior**
No `return in a defer function has no effect`

**Desktop (please complete the following information):**
 - OS: macOS
 - Version of Go: 1.22

**Additional context**
This issue is similar to #863, which is also the one first on Google, but this issue is not exactly the same.

**Comment by chavacava** [comment]
Hi @rpstw thank you for reporting the problem.
I've created a PR #1030 with a proposal of a fix. Please let me know if the PR solves the problem (I've added your code example to the rule's test cases)

**Comment by rpstw** [comment]
Fixed. 🚀
